corehq/apps/receiverwrapper/rate_limiter.py

```diff
@@ -0,0 +1,30 @@
+from corehq.project_limits.rate_limiter import RateDefinition, RateLimiter, \
+    PerUserRateDefinition
+
+# Danny promised in an Aug 2019 email not to enforce limits that were lower than this.
+# If we as a team end up regretting this decision, we'll have to reset expectations
+# with the Dimagi NDoH team.
+rates_promised_not_to_go_lower_than = RateDefinition(
+    per_week=115,
+    per_day=23,
+    per_hour=3,
+    per_minute=0.07,
+    per_second=0.005,
```

Comment:
I don't understand how this makes sense. How can your 1s counter make sense of this value? I don't think you can rate limit below the 1/s.

Reply:
Yeah so this is what the `floor_for_small_domain` is about. The intention here is that (using `rates_promised_not_to_go_lower_than` and `floor_for_small_domain ` as an example, even though we're starting out with using `rates_promised_not_to_go_lower_than.times(2)`) the total effective limits of a domain would follow the table below for different numbers of users

| # of users | second | minute | hour | day | week
| --- | --- | --- | --- | --- | --- |
| 1 | 1 | 10 | 30 | 50 | 115 |
| 10 | 1 | 10 | 30 | 230 | 1150 |
| 100 | 1 | 7 | 300 | 2300 | 11500 |
| 1000 | 5 | 70 | 3000 | 23000 | 115000 |
| 10000 | 50 | 700 | 30000 | 230000 | 1150000 |
| 100000 | 500 | 7000 | 300000 | 2300000 | 11500000 |

Notice in particular how for low numbers it is more like `floor_for_small_domain`, and for high numbers of users, `floor_for_small_domain` becomes irrelevant and it just grows linearly according to `rates_promised_not_to_go_lower_than`.

Reply:
OK, that's what I thought but it still didn't make a lot of sense except purely for 'documentation' purposes. Fine to leave it though.